docs/usage.md

```diff
@@ -44,87 +63,196 @@ TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,
+| `fastq_2` | Full path to FastQ file for Illumina short reads 2. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".                                                             |
 
 An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.
 
+### Genomes input
+
+Magmap needs two mandatory input to run. One, the samplesheet, was explained above while the second, genome input needs to be specified with the option `--genomeinfo`. The file is a `.csv` file and it requires three columns: accno, genome_fna, genome_gff. it looks as follow:
+
+```csv title="samplesheet.csv"
+accno,genome_fna,genome_gff
+GCA_002688505,./genomes/GCA_002688505.fna,./genomes/GCA_002688505.gff
+GCA_002688515,./genomes/GCA_002688515.fna,
+```
+
+N.B.: you don't need to provide gff, Prokka can handle if you only provide fasta files but we recommend when adding gff that the file is generated by using prokka in order to don't get any conflict with other generated gffs.
+
+### Other inputs (optional)
+
+Magmap can handle several types of input that can be used for different purpose.
+
+#### Indexes input
+
+The indexes input is used by Sourmash to select genomes that can downloaded in a second step and added to the pipeline.
+It is provided with the `--indexes` parameter and it is a path (local or remote).
+
+```
+https://github.com/nf-core/test-datasets/raw/magmap/testdata/sourmash_test.index.sbt.zip
+```
+
+N.B.: the sbt files can be generated with sourmash (check it [here](https://sourmash.readthedocs.io/en/latest/index.html)) or some can be found in sourmash documentation website.
+E.g. "sourmash gtdb sbt" you will find the indexes for gtdb.
+
+#### metadata input
+
+Magmap accepts several metadata as `.csv` files that provides information about the genomes that you will use in the pipeline.
+
+##### gtdb_metadata
+
+This file contains several information that can be found in gtdb metadata on the official [website](https://data.ace.uq.edu.au/public/gtdb/data/releases/release80/80.0/). You can either use directly the same file found there or you can make a custom one.
+
+##### gtdb-tk metadata
+
+This file contains several information that can be obtained as gtdb-tk output. You can either use directly the same file found there or you can make a custom one.
+
+##### checkM metadata
+
+This file contains several information that can be obtained as checkM output. You can either use directly the same file found there or you can make a custom one.
+
+### Check duplicates
+
+The pipeline is checking if there are any duplicate names among the genomes that the user provides. If there are duplicates, the pipeline will stop and return a file with the contig names that needs to be changed in their name in order to work. This is done to avoid overlapping in the following steps (e.g. same prokka output for the protein sequences and the gffs).
```

Comment:
```suggestion
The pipeline will perform validation checks to see if there are any duplicate names among the genomes that the user provides. If there are duplicates, the pipeline will stop and return a file with the contig names that needs to be changed in their name in order to work. This is done to avoid overlapping in the following steps (e.g. same prokka output for the protein sequences and the gffs).
```